csm/cli/cortxcli.py

```diff
@@ -90,21 +90,25 @@ def login(self):
         :return:None
         """
         try:
-            self.username = input('Username: ').strip()
-            Log.debug(f"{self.username} attempted to Login.")
-            if self.username:
+            if not self.extract_token():
+                self.username = input('Username: ').strip()
+                Log.debug(f"{self.username} attempted to Login.")
+                if not self.username:
+                    self.do_exit("Username wasn't provided.")
                 password = getpass(prompt="Password: ")
                 self._session_token = self.loop.run_until_complete(self.rest_client.login(
                     self.username, password))
                 if not self._session_token:
                     self.do_exit("Server authentication check failed.")
-                self.headers = {'Authorization': f'Bearer {self._session_token}'}
-                Log.info(f"{self.username}: Logged in.")
-                response = self.loop.run_until_complete(self.rest_client.permissions(self.headers))
-                if response:
-                    self._permissions.update(response)
-            else:
-                self.do_exit("Username wasn't provided.")
+            self.headers = {'Authorization': f'Bearer {self._session_token}'}
```

Comment:
In case of PAM : self._session_token might not get populated if some file error occurs. It is better to check first and raise an error.

Reply:
@pawankumarsrivastava, If anything goes wrong in extract_token(), we are returning False. So in this case user will be prompted for 'username' & 'password'  and authentication will happen, self._session_token will get populated from there. And if authentication fails, we are handling it.

Reply:
What if you have read from file and deletion fails? You have the session with you that you read from file and since you returned false you will ask for username and password . So, you will have two calls even you have the session

Reply:
@pawankumarsrivastava , In the last commit, I have changed this behavior. In case of file deletion fails, I'm logging the error and **not returning false**.  So that, user should be able to login if self._session_token is populated and it is valid.